AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -177,16 +173,16 @@ - (void)generatePlaceholderImage
 - (void)imageAtTime:(CMTime)imageTime completionHandler:(void(^)(UIImage *image))completionHandler
 {
   ASPerformBlockOnBackgroundThread(^{
-    ASDN::MutexLocker l(_videoLock);
+    AVAsset *asset = self.asset;
```

Comment:
Moving `imageAtTime` out of `ASVideoNode` is implemented here: https://github.com/facebook/AsyncDisplayKit/compare/master...ejensen:video-ui-blocking Please let me know if you feel that makes the data consitency logic more or less clear. Any other suggestions are more than welcome.